addons/website/models/res_lang.py

```diff
@@ -22,16 +22,30 @@ def _get_frontend(self) -> LangDataDict:
         """
         if request and getattr(request, 'is_frontend', True):
             lang_ids = self.env['website'].get_current_website().language_ids.sorted('name').ids
-            ResLang = self.env['res.lang']
-            langs = [dict(ResLang._get_data(id=id_)) for id_ in lang_ids]
-            # if only one region for a language, use only the language code
-            shorts = [lang['code'].split('_')[0] for lang in langs]
-            for lang, short in zip(langs, shorts):
-                if shorts.count(short) == 1:
-                    lang['hreflang'] = short
+            langs = [dict(self.env['res.lang']._get_data(id=id_)) for id_ in lang_ids]
+            es_419_exists = any(lang['code'] == 'es_419' for lang in langs)
+            already_shortened = []
+            for lang in langs:
+                code = lang['code']
+                short_code = code.split('_')[0]
+                # Always shorten one language for each group of languages.
+                # Special case for spanish, as es_419 is not a valid hreflang
+                # and es_419 is actually the new "generic" spanish, when it is
+                # in the available languages, it should be the one shortened.
+                if (
+                    short_code not in already_shortened
```

Comment:
It will force the first lang to be the short hreflang. 
Small limitation, but fixing other case like if you install zh_cn zh_tw where none of them are shortened.